src/LogicTasks/Syntax/DecomposeFormula.hs

```diff
@@ -0,0 +1,143 @@
+import Data.Maybe (isNothing, fromJust)
+import Trees.Helpers (collectLeaves, collectUniqueBinOpsInSynTree, swapKids)
+import Data.Containers.ListUtils (nubOrd)
+import Control.Monad.Cont (MonadIO (liftIO))
+import LogicTasks.Syntax.TreeToFormula (cacheTree)
+
+
+
+
+description :: OutputMonad m => DecomposeFormulaInst -> LangM m
+description DecomposeFormulaInst{..} = do
+
+    example (display tree) $ do
+      english "The following formula is given:"
+      german "Die folgende Formel ist gegeben:"
+
+    instruct $ do
+      english "Create the syntax tree for this formula and swap the two subtrees at the root. "
+      english "Finally, enter the formula for the resulting syntax tree."
+      german "Erstellen Sie den Syntaxbaum zu dieser Formel und tauschen Sie die beiden Teilbäume an der Wurzel. "
+      german "Geben Sie anschließend die Formel für den entstandenen Syntaxbaum ein."
+
+    instruct $ do
+      english "(You are allowed to add arbitrarily many additional pairs of brackets in the formula.)"
+      german "(In der Formel dürfen Sie beliebig viele zusätzliche Klammerpaare hinzufügen.)"
+
+    when addExtraHintsOnAssociativity $ instruct $ do
+      english "Remark: Do not try to use associativity in this task in order to save brackets."
+      german "Hinweis: Sie dürfen bei dieser Aufgabe nicht Klammern durch Verwendung von Assoziativität einsparen."
+
+    keyHeading
+    basicOpKey
+
+    paragraph $ indent $ do
+      translate $ do
+        english "A solution attempt could look like this: "
+        german "Ein Lösungsversuch könnte beispielsweise so aussehen: "
+      translatedCode $ flip localise $ translations $ do
+        english "(A or not B) and C"
+        german "(A oder nicht B) und C"
+      pure ()
+
+    extra addText
+    pure ()
+
+
+verifyInst :: OutputMonad m => DecomposeFormulaInst -> LangM m
+verifyInst _ = pure ()
+
+
+
+verifyConfig :: OutputMonad m => DecomposeFormulaConfig -> LangM m
+verifyConfig = checkDecomposeFormulaConfig
+
+
+
+start :: TreeFormulaAnswer
+start = TreeFormulaAnswer Nothing
+
+
+
+partialGrade :: OutputMonad m => DecomposeFormulaInst -> TreeFormulaAnswer -> LangM m
+partialGrade DecomposeFormulaInst{..} sol = do
+
+  when (isNothing solTree) $ reject $ do
+    english "You did not submit a solution."
+    german "Die Abgabe ist leer."
+
+  when (any (`notElem` origLiterals) solLiterals) $ reject $ do
+    english "Your solution contains unknown literals."
+    german "Ihre Abgabe beinhaltet unbekannte Literale."
+
+  unless (length origLiterals == length solLiterals) $ reject $ do
+    english "Your solution does not contain all literals present in the original formula."
+    german "Ihre Abgabe beinhaltet nicht alle Literale aus der ursprünglichen Formel."
+
+  unless (length origOperators == length solOperators) $ reject $ do
+    english "Your solution contains too many different operators."
+    german "Ihre Abgabe beinhaltet zu viele unterschiedliche Operatoren."
```

Comment:
That seems off. The check is for them not being equal, but the message claims that one side is larger than the other. But not being equal could also mean that side is actually smaller than the other.

Reply:
I updated the wording.